
<launch>
  
  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find rl_navigation)/config/lidar.rviz"/>  
  
  <node pkg="urg_node" type="urg_node" name="urg_node" respawn="true">
    <param name="serial_port" value="/dev/serial/by-id/usb-Hokuyo_Data_Flex_for_USB_URG-Series_USB_Driver-if00" type="str" />
    <!--param name="serial_port" value="/dev/ttyACM2" type="str" /-->
    <param name="frame_id" value="laser" type="str" />
    <param name="calibrate_time" value="true" type="bool" />
    <param name="publish_intensity" value="false" type="bool" />
    <param name="publish_multiecho" value="false" type="bool" />
    <param name="angle_min" value="-1.5657963" type="double" />
    <param name="angle_max" value="1.5707963" type="double" />
  </node>
  
  <node name="input_publisher" pkg="rl_navigation" type="input_publisher.py" output="screen">
    <!--neural network input. -->
    <!--default:depth×36, goal_info×2 (total:38)-->
    <param name="input_depth_n" value="19"/>
    <param name="visual_odom" value="False"/>
    <param name="human_detection" value="False"/>
  </node>

  <include file="$(find ypspur_ros)/launch/ypspur_ros.launch"/>

</launch>
